packages/api/src/command/medical/cq-directory/create-cq-directory-entry.ts

```diff
@@ -18,15 +20,72 @@ export const createOrUpdateCQDirectoryEntry = async (
     return { org: updOrg, updated: true };
   }
 
-  const org = await createDirectoryDirectoryEntry(orgData);
+  const org = await createCQDirectoryEntry(orgData);
   return { org, updated: false };
 };
 
-async function createDirectoryDirectoryEntry(
-  orgData: CQDirectoryEntryData
-): Promise<CQDirectoryEntry> {
+async function createCQDirectoryEntry(orgData: CQDirectoryEntryData): Promise<CQDirectoryEntry> {
   return await CQDirectoryEntryModel.create({
     id: uuidv7(),
     ...orgData,
   });
 }
+
+export const createOrUpdateCQDirectoryEntries = async (
+  orgDataArray: CQDirectoryEntryData[]
+): Promise<CQOrganizationCreateResponse[]> => {
+  const oids = orgDataArray.map(data => data.oid);
+  const existingEntries = await getCQDirectoryEntriesByOids(oids);
```

Comment:
So far, I took into account #2, 4, 5 and 6. 

#1 I marked as a TODO, and for #3, we only iterate thru it once.

Reply:
> #1 I marked as a TODO

As in, this PR's TODO, or TODO for later/another PR?

Also, I forgot about this:
7. Is it worth to run check for existing entries and update them selectively, or should we just update it all? If we always recreate, we save (1) loading data (which I believe is like +99% of the records, right?) and (2) comparing it in memory (which will not be cheap, very CPU intensive considering the amount of data).
8. Do we need the directory's ID? If not, removing the PK also removes the associated index, which will speed up a lot building it.
9. We should prob remove indexes before we recreate the directory and add them afterwards, this should speed up a lot the whole process.

Reply:
Later TODOs. I will update the description and create new issues to link here. 

For #7, I think it's worth only updating the entries that need an update. I think at some point, when we start querying the directory on a daily basis, we will probably want to know which organizations actually got updated, so we can re-query for new organization x patient associations, especially in the case of an address change. 

For #8, the directory ID is the OID of the organization. 

For #9, I'll look into it, but might need clarifications - will take it to Slack

Reply:
7) That's a good point, but I would make sure that's the case, because I assume the difference in execution time from always create vs. update to be significant. If we need to update, could look into `.upsert()` so we don't need to load and compare.

8) Gotcha. And are we going to query by OID? Or always by lat/lon?

Reply:
8) probably OID and point, rather than lat/lon. Maybe we should add another index for the point?

Reply:
> Maybe we should add another index for the point?

If we're going to query the directory using that as the main filter, likely yes. Would need to look at the filter/conditions to be able to better recommend what to do here. We can add it when we PR this query.